### PR TITLE
Create facility list items table interactions

### PR DESCRIPTION
**Pull request by kellyi**
## Overview

This PR accomplishes a few things:

- updates the API to handle confirming or rejecting potential facility matches
- adds tests for API confirm/reject logic
- updates the UI to add several interactions to the facility list items list table, including:
   - showing toggleable detail rows for list items in the matched or error states, along with additional details
    - showing a confirm/reject match UI along with a confirmation dialog to use for confirming or rejecting matches
- adds links to a user's facilities on the main list page and also in the navigation bar dropdown
- update the match functionality to create some random matches if the list item id % 4 === 0. This is done because the previous naive matching function was setting everything with 100% confidence

Connects #192

## Demo

confirm-match

reject-matches

## Testing

- get this branch
- resetdb
- serve the app, then register for and log in with a new account
- upload a list from the set of lists committed to the repo (pick one of the larger ones to ensure lots of matches)
- in the VM, run the following commands to process the list:

```
./scripts/manage batch_process --action parse --list-id 15
./scripts/manage batch_process --action geocode --list-id 15
./scripts/manage batch_process --action match --list-id 15
```

- visit your lists page in the UI, then click on the list to go to the details page
- try the following interactions:
   - clicking on an ERROR or MATCH or CONFIRMED_MATCH row should expand that row to display details, and close all other non-CONFIRM details rows
   - clicking on an expanded non-CONFIRM detail row should cause it to collapse back to a normal table row
    - confirming a match through the UI should update that table row to display it as a CONFIRMED_MATCH with a link to the facility that has matched
    - rejecting one of several matches should keep the confirm row open, but show the rejected match as rejected
    - rejecting the last potential match for a list item should create a new facility then update the details row to show it in CONFIRMED_MATCH state

- run `./scripts/test` to verify that they pass

### CODE REVIEW

**Comment by kellyi** [comment]
Just so I'll have it ready at hand, here's the command for parsing the first list uploaded after the initial fixture data is in:

./scripts/manage batch_process --action parse --list-id 15

File: src/django/api/tests.py [hunk 1]
```diff
@@ -181,7 +181,8 @@ def test_cant_replace_twice(self):
         self.assertEqual(response.status_code, status.HTTP_400_BAD_REQUEST)
         self.assertEqual(
             json.loads(response.content),
-            ['FacilityList 1 has already been replaced.'])
+            ['FacilityList {} has already been replaced.'
+                .format(response_json['id'])])
```
Comment (kellyi) [inline src/django/api/tests.py]: Not sure why this was breaking, but it kept trying to use "FacilityList 15 has already been", which made it seem like it was uploading the new list after loading the non-test fixture data.

**Comment by jwalgran** [comment]
Looking at this now

File: src/app/src/components/FacilityListItemsMatchTableRow.jsx [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+            />
+        </TableCell>
+        <TableCell
+            padding="none"
+            style={listTableCellStyles.nameCellStyles}
+        >
+            <FacilityListItemsDetailedTableRowCell
+                title={name || ' '}
+                subtitle="Facility Match Name"
+                hsIsHidden={false}
```
Comment (jwalgran) [inline src/app/src/components/FacilityListItemsMatchTableRow.jsx]: Typo? hs -> hr?

File: src/app/src/components/FacilityListItemsConfirmationTableRow.jsx [hunk 3]
```diff
@@ -0,0 +1,174 @@
...
+                />
+            </TableCell>
+            <TableCell
+                padding="none"
+                style={listTableCellStyles.nameCellStyles}
+            >
+                <FacilityListItemsDetailedTableRowCell
+                    title={item.name || ' '}
+                    subtitle="Matched Name"
+                    hsIsHidden={false}
```
Comment (jwalgran) [inline src/app/src/components/FacilityListItemsConfirmationTableRow.jsx]: Typo? hs -> hr?

File: src/app/src/components/FacilityListItemsConfirmationTableRow.jsx [hunk 4]
```diff
@@ -0,0 +1,174 @@
...
+                />
+            </TableCell>
+            <TableCell
+                padding="none"
+                style={listTableCellStyles.statusCellStyles}
+            >
+                <FacilityListItemsDetailedTableRowCell
+                    title={item.status}
+                    subtitle="Actions"
+                    hsIsHidden={false}
```
Comment (jwalgran) [inline src/app/src/components/FacilityListItemsConfirmationTableRow.jsx]: Typo? hs -> hr?

File: src/app/src/components/FacilityListItemsMatchTableRow.jsx (hunk 2 above)
Comment (kellyi) [reply src/app/src/components/FacilityListItemsMatchTableRow.jsx]: Excellent catch! Will squash in a fixup to address.

**Comment by kellyi** [comment]
squashed in a fixup to replace the `hs` prefix with `hr`, then tested everything out again. Everything's still working so will merge once the tests pass.

Thanks for your help with this!

**Pull request merged**